validate that the node is rented if gpu is enabled

0oM4R opened the pull request:
### Update
- revalidate the selected node after reserve/unreserve node 
- move the validate rent contract and checkCapacity pool to be at the end of the validation task to give the priority to the resources validation

### Description

![image](https://github.com/user-attachments/assets/ed5d11ac-6177-448a-834f-ab6510df9916)

### Changes

Add the validation on the `validateRentContracts` method as it is a common validation between auto and manual selection 
also move the rented validation to be only inside the validateRentContracts in manual selection to enable deploying on dedicated nodes on shared farms 
![image](https://github.com/user-attachments/assets/6ee9b88a-7884-4625-bd9a-337b0fc16fae)

### Related Issues

- #3944 
- #3951
### Tested Scenarios

- enabled the GPU switch then select a node 
- entered a node id in manual selection with GPU switch is enabled 

### Documentation PR

For UI changes, Please provide the Documentation PR on [info_grid](https://github.com/threefoldtech/info_grid)

### To consider

#### Preliminary Checks:

- [x] Preliminary Checks
  - Does it completely address the issue linked?
  - What about edge cases?
  - Does it meet the specified acceptance criteria?
  - Are there any unintended side effects?
  - Does the PR adhere to the team's coding conventions, style guides, and best practices?
  - Does it integrate well with existing features?
  - Does it impact the overall performance of the application?
  - Are there any bottlenecks or slowdowns?
  - Has it been optimized for efficiency?
  - Has it been adequately tested with unit, integration, and end-to-end tests?
  - Are there any known defects or issues?
  - Is the code well-documented?
  - Are changes to documentation reflected in the code?

#### UI Checks:

- [x] UI Checks
  - If a UI design is provided/ does it follow it?
  - Does every button work?
  - Is the data displayed logical? Is it what you expected?
  - Does every validation work?
  - Does this interface feel intuitive?
  - What about slow network? Offline?
  - What if the gridproxy/graphql/chain is failing?
  - What would a first time user have a hard time navigating here?

#### Code Quality Checks:

- [x] Code Quality Checks
  - Code formatted/linted? Are there unused imports? .. etc
  - Is there redundant/repeated code?
  - Are there conditionals that are always true or always false?
  - Can we write this more concisely?
  - Can we reuse this code? If yes, where?
  - Will the changes be easy to maintain and update in the future?
  - Can this code become too complex to understand for other devs?
  - Can this code cause future integration problems?

### Testing Checklist

- [ ] Does it Meet the specified acceptance criteria.
- [ ] Test if changes can affect any other functionality.
- [ ] Tested with unit, integration, and end-to-end tests.
- [ ] Tested the un-happy path and rollback scenarios.
- [ ] Documentation updated to meet the latest changes.
- [ ] Check automated tests working successfully with the changes.
- [ ] Can be covered by automated tests.

### General Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstring
- [x] Screenshots/Video attached (needed for UI changes)

0oM4R committed:
feature:[TFNodeSelector]: validate that the node is rented if gpu is selected

0oM4R committed:
Refactor:[TFManualSelector]: validate that the node is rented inside the validaterentContract function

amiraabouhadid commented:
tested flow in automated node selection, toggled GPU switch, 
error was correctly shown when gpu node selected was not rented. Validation err was gone once node has been rented. Was able to successfully deploy. 
![image](https://github.com/user-attachments/assets/44f678c0-07eb-4661-8569-b96026f54152)

But in manual selection, err does not go away even after a while of node being rented 
![image](https://github.com/user-attachments/assets/7f62381a-45dc-4f3f-8665-5d2062568966)

0oM4R committed:
Feature[ManualSelector]:
- revalidate the node after reserve/unreserve selected node
- move the validate rent contract to be the last validation step after checkNodeCapacityPool

0oM4R commented:
> But in manual selection, err does not go away even after a while of node being rented
> ![image](https://private-user-images.githubusercontent.com/56790126/418336907-7f62381a-45dc-4f3f-8665-5d2062568966.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA5MDk0NTYsIm5iZiI6MTc0MDkwOTE1NiwicGF0aCI6Ii81Njc5MDEyNi80MTgzMzY5MDctN2Y2MjM4MWEtNDVkYy00ZjNmLTg2NjUtNWQyMDYyNTY4OTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzAyVDA5NTIzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3NDE5MzkzODM0Y2RhYzMyZTljY2FiYTQ1Njk4NWUyZjc3YmQ5YTE1N2U2ZmY1MTg1ZWE1ZGM3ZDc1MDk1YmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5LLGPZU6RJJRxwUAkASvANYfH-Fsf4ixbwNU7P5Nqrk)

That was mentioned in another issue however, its solution is ready on the pr rn

amiraabouhadid reviewed (approved):
great, err in manual selection goes away now upon node rental 
was able to successfully deploy
